File: lessons/compiled_lang/rust/src/d00.input.rs
use std::io::{self, Write};

fn main() {
    print!("Enter your name: ");
    io::stdout().flush().unwrap();

    let mut name = String::new();
    io::stdin().read_line(&mut name).unwrap();
    let name = name.trim_end();

    println!("Hello {name}!");
}

File: lessons/compiled_lang/java/src/d00.input.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class D00Input {
    public static void main(String[] args) throws IOException {
        BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Enter your name: ");
        String name = stdin.readLine();
        System.out.println("Hello " + name + "!");
    }
}
